### PR TITLE
feat: Support security token for OSS artifact driver

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 1]
```diff
@@ -2118,6 +2118,9 @@ type OSSBucket struct {
 
 	// CreateBucketIfNotPresent tells the driver to attempt to create the OSS bucket for output artifacts, if it doesn't exist
 	CreateBucketIfNotPresent bool `json:"createBucketIfNotPresent,omitempty" protobuf:"varint,5,opt,name=createBucketIfNotPresent"`
+
+	// SecurityToken is the user's temporary security token
```
Comment (alexec) [inline pkg/apis/workflow/v1alpha1/workflow_types.go:2122]: could we add a link to documentation please?

**Review by alexec — CHANGES_REQUESTED** [review]

**Comment by alexec** [comment]
@jessesuen can we get approval to these proto changes?

File: pkg/apis/workflow/v1alpha1/workflow_types.go (hunk 1 above)
Comment (terrytangyuan) [reply pkg/apis/workflow/v1alpha1/workflow_types.go]: Sure. Added.

**Comment by terrytangyuan** [comment]
Hi @jessesuen! In case you missed it among other notifications, could you review this simple proto change when you get a chance? Thanks!

**Review by alexec — APPROVED** [review]

**Pull request merged**